consensus_test.go

```diff
@@ -11,287 +15,754 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/trace"
 )
 
-func TestTransition_ValidateState_Prepare(t *testing.T) {
-	t.Skip()
+var (
+	mockProposal  = []byte{0x1, 0x2, 0x3}
+	mockProposal1 = []byte{0x1, 0x2, 0x3, 0x4}
+)
+
+// MockedTracer is a mocked object that implements Tracer interface
+type MockedTracer struct {
+	mock.Mock
+}
+
+func (m *MockedTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	args := m.Called(ctx, spanName, opts)
+	return args.Get(0).(context.Context), args.Get(1).(trace.Span)
+}
+
+func Test_InitializeConfigOptionsSet(t *testing.T) {
+	t.Parallel()
+
+	pool := newTesterAccountPool()
+	pool.add("A", "B", "C")
 
+	m := &mockPbft{
+		t:        t,
+		pool:     pool,
+		respMsg:  []*MessageReq{},
+		sequence: 1, // use by default sequence=1
+	}
+
+	timeout := 1 * time.Millisecond
+	proposalTimeout := 2 * time.Millisecond
+	mockedTracer := new(MockedTracer)
+	mockedTimeout := func(u uint64) time.Duration { return 0 }
+
+	assert.NotPanics(t, func() {
+		m.Pbft = New(pool.get("A"), m,
+			WithTimeout(timeout),
+			WithProposalTimeout(proposalTimeout),
+			WithTracer(mockedTracer),
+			WithRoundTimeout(mockedTimeout))
+	}, "New function failed to execute")
+	assert.Same(t, mockedTracer, m.Pbft.config.Tracer)
+
+	assert.Equal(t, timeout, m.Pbft.config.Timeout)
+	assert.Equal(t, proposalTimeout, m.Pbft.config.ProposalTimeout)
+	assert.Equal(t, mockedTracer, m.Pbft.config.Tracer)
+
+	mockedTracer.AssertNotCalled(t, "Start")
+	mockedTracer.AssertExpectations(t)
+}
+
+func TestTransition_AcceptState(t *testing.T) {
```

Comment:
Please revert this change. It is fine but when you are debugging or changing parts of the consensus and want to test specific functions, it is cumbersome to have to comment out this array instead of running ```go test -v ./... -run TestAcceptState_Something```.